### PR TITLE
allow for json/ndjson content type with charset

### PR DESCRIPTION
**Pull request by aspacca**
Enhancement
## Why is it important?
see: https://github.com/elastic/enhancements/issues/16194

## Checklist

- [x] My code follows the style guidelines of this project
~- [ ] I have commented my code, particularly in hard-to-understand areas~
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- 

Closes https://github.com/elastic/enhancements/issues/16194

### CODE REVIEW

File: x-pack/filebeat/input/awss3/input_integration_test.go [hunk 1]
```diff
@@ -328,11 +325,19 @@ func uploadS3TestFiles(t *testing.T, region, bucket string, filenames ...string)
 			t.Fatalf("Failed to open file %q, %v", filename, err)
 		}
 
+		contentType := ""
+		if strings.HasSuffix(filename, "ndjson") || strings.HasSuffix(filename, "ndjson.gz") {
+			contentType = "let-CI-fail-" + contentTypeNDJSON + "; charset=UTF-8"
```
Comment (tommyers-elastic) [inline x-pack/filebeat/input/awss3/input_integration_test.go:330]: what is the reason for this `let-CI-fail`?

Comment (aspacca) [reply x-pack/filebeat/input/awss3/input_integration_test.go:330]: I wanted to asses that the integration tests were run in CI, forcing a failure

they don't: CI is green anyway 

(having `let-CI-fail-` has suffix https://github.com/elastic/beats/pull/32767/files#diff-f345fd6a1f5ea9523117d4ead2e5f1d13fb82eb1c65a089fd34fcdd514916a96R156 will be false)

**Review by kaiyan-sheng — APPROVED** [review]
LGTM

**Comment by kaiyan-sheng** [comment]
Do we also need to backport it to 7.17?

**Comment by aspacca** [comment]
> Do we also need to backport it to 7.17?

I didn't backport to 7.17 since it's a difference code base now (aws sdk v2 upgrade)

I will create a new PR for that, thanks for reminding me

**Comment by aspacca** [comment]
> I didn't backport to 7.17 since it's a different code base now (aws sdk v2 upgrade)

it seems the lines affected didn't change, so I will add the backport label here

**Pull request merged**